editor/views/editoritem.py

```diff
@@ -511,10 +511,51 @@ def get(self, *args, **kwargs):
 
     def post(self, request, *args, **kwargs):
         ei = self.get_object()
+        error_message = ''
+        count = 1
+        if (ei.item_type == 'exam'):
+            exam_questions = ei.exam.newexamquestion_set.all().order_by('qn_order').distinct()
+            for eq in exam_questions:
+                question = eq.question
+                if question.editoritem.published == True:
+                    continue
+                current_error_message = ''
+                if not question.editoritem.can_be_edited_by(self.request.user):
+                    current_error_message = 'User cannot edit this question. '
+                edit_content = question.editoritem.edit_dict()['JSONContent']
+                if edit_content['statement'] == '':
+                    current_error_message = current_error_message + 'No statement provided. '
+                parts = edit_content['parts']
+                if len(parts) == 0:
+                    current_error_message = current_error_message + 'No parts provided. '
+                variables = edit_content['variables']
+                if len(variables) == 0:
+                    current_error_message = current_error_message + 'No variables given. '
+                advice = edit_content['advice']
+                if len(advice) == 0:
+                    current_error_message = current_error_message + 'No advice given. '
+                name = edit_content['name']
+                if name == '':
+                    current_error_message = current_error_message + 'No name for question. '
+                settings = edit_content['metadata']
+                license_rights = settings['licence']
+                description = settings['description']
+                if license_rights == 'None specified':
+                    current_error_message = current_error_message + 'No license selected. '
+                if len(description) == 0:
+                    current_error_message = current_error_message + 'No description provided. '
+                
+                if len(current_error_message) > 0:
+                    error_message = error_message + 'Cannot publish question ' + str(count) + ': ' + name + '. ' + current_error_message + '\n'
```

Comment:
Picking out just one of the potentially many reasons for not publishing a question isn't particularly helpful. We could show a list of things that need to be done for each question, but that's quite a lot of information. 
Something like 

> The following questions need more information before they can be automatically published: 
> 
> * 1: **name of question 1**
> * 4: **name of question 4**
> 
> Click on each question to see what needs to be done.